ocaml/xapi/xapi_session.ml

```diff
@@ -1337,3 +1349,73 @@ let create_from_db_file ~__context ~filename =
   in
   let db_ref = Some (Db_ref.in_memory (ref (ref db))) in
   create_readonly_session ~__context ~uname:"db-from-file" ~db_ref
+
+let get_all_records ~__context =
+  let get_api_record ~self =
+    {
+      API.session_auth_user_name= Db.Session.get_auth_user_name ~__context ~self
+    ; API.session_auth_user_sid= Db.Session.get_auth_user_sid ~__context ~self
+    ; API.session_client_certificate=
+        Db.Session.get_client_certificate ~__context ~self
+    ; API.session_is_local_superuser=
+        Db.Session.get_is_local_superuser ~__context ~self
+    ; API.session_last_active= Db.Session.get_last_active ~__context ~self
+    ; API.session_originator= Db.Session.get_originator ~__context ~self
+    ; API.session_other_config= Db.Session.get_other_config ~__context ~self
+    ; API.session_parent= Db.Session.get_parent ~__context ~self
+    ; API.session_pool= Db.Session.get_pool ~__context ~self
+    ; API.session_rbac_permissions=
+        Db.Session.get_rbac_permissions ~__context ~self
+    ; API.session_subject= Db.Session.get_subject ~__context ~self
+    ; API.session_tasks= Db.Session.get_tasks ~__context ~self
+    ; API.session_this_host= Db.Session.get_this_host ~__context ~self
+    ; API.session_this_user= Db.Session.get_this_user ~__context ~self
+    ; API.session_uuid= Db.Session.get_uuid ~__context ~self
+    ; API.session_validation_time=
+        Db.Session.get_validation_time ~__context ~self
+    ; API.session_creation_ip= Db.Session.get_creation_ip ~__context ~self
+    ; API.session_last_login_ip= Db.Session.get_last_login_ip ~__context ~self
+    }
+  in
+  Db.Session.get_all ~__context
+  |> List.map (fun session -> (session, get_api_record ~self:session))
+
+let count_sessions ~__context =
+  Db.Session.get_all ~__context |> List.length |> Int64.of_int
+
+let group_by_originator ~__context =
+  let one = Int64.of_int 1 in
+  Db.Session.get_all ~__context
+  |> List.map (fun self -> Db.Session.get_originator ~__context ~self)
+  |> List.sort (fun orig1 orig2 -> if orig1 > orig2 then 1 else 0)
```

Comment:
This compare function never returns -1, as it probably should. You can probably use `List.sort String.compare`

Reply:
That would work well